src/taipy/core/_repository/_v2/_sql_repository.py

```diff
@@ -35,28 +32,35 @@ def __init__(self, model: Type[ModelType], converter: Type[Converter], db_table)
         * `db_table`: A SQLAlchemy model class
         """
         self.model = model
-        self.db = SessionLocal()
+        self.db = session
         self.converter = converter
 
+    ###############################
+    # ##   Inherited methods   ## #
+    ###############################
     def _save(self, entity: Entity):
         obj = self.converter._entity_to_model(entity)
-        if entry := self.db.query(self.model).filter_by(model_id=obj.id).first():
+        if entry := self.db.query(self.model).filter_by(id=obj.id).first():
             self.__update_entry(entry, obj)
             return
         self.__insert_model(obj)
 
     def _load(self, model_id: Any) -> Optional[ModelType]:
```

Comment:
`model.id` and `entity.id` both represent the same concept: the identifier of the entity.

I would remove model_id everywhere and use `entity_id` instead, even if we compare it to the `model.id`. So we don't create confusion, and we know that there is only one id: the entity's id. 

```suggestion
    def _load(self, entity_id: Any) -> Optional[ModelType]:
```